plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+		}
+		r.Client = &http.Client{
+			Transport: tr,
+			Timeout:   r.ClientTimeout.Duration,
+		}
+	}
+	return nil
+}
+
+func (r *RavenDB) requestJSON(u string, target interface{}) error {
+	if r.URL == "" {
+		r.URL = DefaultURL
+	}
+	u = fmt.Sprintf("%s%s", r.URL, u)
+
+	req, err := http.NewRequest("GET", u, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := r.Client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if nil != r.Log {
+		r.Log.Debugf("%s: %s", u, resp.Status)
+	}
+	if resp.StatusCode >= 400 {
+		return fmt.Errorf("invalid response code to request '%s': %d - %s", r.URL, resp.StatusCode, resp.Status)
+	}
+
+	json.NewDecoder(resp.Body).Decode(target)
+
+	return nil
+}
+
+func gatherServer(r *RavenDB, acc telegraf.Accumulator) {
+	if !r.GatherServerStats {
+		return
+	}
+
+	serverResponse := &serverMetricsResponse{}
+
+	err := r.requestJSON("/admin/monitoring/v1/server", &serverResponse)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+
+	tags := map[string]string{"url": r.URL}
+	tags["node_tag"] = serverResponse.Cluster.NodeTag
+	tags["cluster_id"] = serverResponse.Cluster.Id
+
+	if serverResponse.Config.PublicServerUrl != "" {
+		tags["public_server_url"] = serverResponse.Config.PublicServerUrl
+	}
+
+	fields := map[string]interface{}{
+		"server_version":                                    serverResponse.ServerVersion,
+		"server_full_version":                               serverResponse.ServerFullVersion,
+		"uptime_in_sec":                                     serverResponse.UpTimeInSec,
+		"server_process_id":                                 serverResponse.ServerProcessId,
+		"config_server_urls":                                strings.Join(serverResponse.Config.ServerUrls, ";"),
+		"backup_max_number_of_concurrent_backups":           serverResponse.Backup.MaxNumberOfConcurrentBackups,
+		"backup_current_number_of_running_backups":          serverResponse.Backup.CurrentNumberOfRunningBackups,
+		"cpu_process_usage":                                 serverResponse.Cpu.ProcessUsage,
+		"cpu_machine_usage":                                 serverResponse.Cpu.MachineUsage,
+		"cpu_processor_count":                               serverResponse.Cpu.ProcessorCount,
+		"cpu_assigned_processor_count":                      serverResponse.Cpu.AssignedProcessorCount,
+		"cpu_thread_pool_available_worker_threads":          serverResponse.Cpu.ThreadPoolAvailableWorkerThreads,
+		"cpu_thread_pool_available_completion_port_threads": serverResponse.Cpu.ThreadPoolAvailableCompletionPortThreads,
+		"memory_allocated_in_mb":                            serverResponse.Memory.AllocatedMemoryInMb,
+		"memory_installed_in_mb":                            serverResponse.Memory.InstalledMemoryInMb,
+		"memory_physical_in_mb":                             serverResponse.Memory.PhysicalMemoryInMb,
+		"memory_low_memory_severity":                        mapMemorySeverity(serverResponse.Memory.LowMemorySeverity),
```

Comment:
Why not using the `string` as is? The string is much more readable than an arbitrary number.

Reply:
that's the reason: https://community.grafana.com/t/is-there-a-text-to-value-mapping/13420 or https://github.com/grafana/grafana/issues/7631

so we can plot this value over time.

Reply:
As I wrote for the bool, you can still use a processor in telegraf for doing the mapping if you need it. This way we can keep the readability and you can still use grafana. Anyway I think it's a bit ugly to do things in the data-collection just to work around a deficit of the visualization. :-)

Reply:
ok I got this: https://github.com/influxdata/telegraf/blob/release-1.17/plugins/processors/enum/README.md#enum-processor-plugin will use enums than

Reply:
Please mark as resolved.